src/inmanta/server/protocol.py

```diff
@@ -130,16 +131,16 @@ def start(self):
             LOGGER.debug("Created REST transport with SSL")
         else:
             self.http_server = HTTPServer(application, decompress_request=True)
-
         self.http_server.listen(port)
 
         LOGGER.debug("Start REST transport")
 
+    @gen.coroutine
     def stop(self):
-        LOGGER.debug("Stoppin Server Rest Endpoint")
+        LOGGER.debug("Stopping Server Rest Endpoint")
         self.http_server.stop()
         for endpoint in self.__end_points:
```

Comment:
stopping in starting order will be a problem if the database connection is properly closed

should be reverse order imho